Feature/vonk 4187 access policy doc (#220)
FirelyTeam/firely-docs-firely-server: security/permissions.rst

Permissions (AccessPolicy)
--------------------------

> Firely Server supports filtering the requested client scopes by using built-in custom access policy resources. Both Smart-V1 and Smart-V2 scopes are supported. 

Reviewer: Firely Server supports filtering the **granted** access scopes for a client by using built-in custom access policy resources.

> Firely Server supports filtering the requested client scopes by using built-in custom access policy resources. Both Smart-V1 and Smart-V2 scopes are supported. 

Reviewer: The access policy decisions are based on HL7 SMART on FHIR scopes, both SMART on FHIR v1 and v2 scopes are supported.

> The resource definitions are preloaded in Firely Server and can be viewed on the administration endpoint API or in Simplier under 

Reviewer: The StructureDefinitions ...

Reviewer: Typo in "Simplier"

> The resource definitions are preloaded in Firely Server and can be viewed on the administration endpoint API or in Simplier under 

Reviewer 2: ```suggestion
The StructureDefinitions are preloaded in Firely Server and can be viewed on the administration endpoint API or in Simplifier under 
```

Reviewer 2: .suo file was added in the last commit. Consider adjusting .gitignore or removing the file

> `Firely Server Definitions - Access Policy <https://simplifier.net/Vonk-ResourcesR4/~resources?text=access&fhirVersion=R4&sortBy=RankScore_desc>`_ .

Reviewer 2: ```suggestion
`Firely Server Definitions - Access Policy <https://simplifier.net/Vonk-ResourcesR4/~resources?text=accesspolicy&fhirVersion=R4&sortBy=RankScore_desc>`_ .
```

> `Firely Server Definitions - Access Policy <https://simplifier.net/Vonk-ResourcesR4/~resources?text=access&fhirVersion=R4&sortBy=RankScore_desc>`_ .

Reviewer 2: When checking the resource on Simplifier, I see a warning for the AccessPolicyDefinition : 
`The canonical from this resource does not match any claim in this context and conflicts with a claim from another scope.

http://hl7.org/fhir

Canonical claims are used to verify ownership of your canonical URLs.`
When I check the URL field I see indeed:
`"url": "http://hl7.org/fhir/StructureDefinition/AccessPolicyDefinition",`
Also, when checking the "AccessPolicy", I see that the `AccessPolicy.instantiatesCanonical` field refers to a target profile of type: `"http://fire.ly/fhir/StructureDefinition/AccessPolicyDefinition"`

I am not that familiar with FHIR, but it does not seem normal.
Could you confirm?

Author: @alexzautke. I am not familiar with Simplifier enough to evaluate this. Do you know?

The *AccessPolicyDefinition* resource controls the scopes which are permissible. 

> The *AccessPolicy* resource contains the references to Patient, Group, Practitioner, PractitionerRole, Person, RelatedPerson and Device for which the AccessPolicyDefinition applies.

Reviewer 2: ```suggestion
The *AccessPolicy* resource associates Patient, Group, Practitioner, PractitionerRole, Person, RelatedPerson and Device with AccessPolicyDefinition(s).
```
